Lib/json/tool.py

```diff
@@ -31,21 +31,26 @@ def main():
     options = parser.parse_args()
 
     infile = options.infile or sys.stdin
-    outfile = options.outfile or sys.stdout
     sort_keys = options.sort_keys
     json_lines = options.json_lines
-    with infile, outfile:
+    with infile:
         try:
             if json_lines:
-                objs = (json.loads(line) for line in infile)
+                objs = tuple(json.loads(line) for line in infile)
```

Comment:
This change is wrong. It is a generate on purpose, to not load all JSON in memory at once.

Reply:
Hi @vstinner, thank you for taking the time to  review this PR!

This is what I was referring to in https://github.com/python/cpython/pull/7865#issuecomment-449981840.

If the file is not fully read before opening `outfile`, the bug in https://bugs.python.org/issue33927 will be there and if you close `infile` before reading it another exception will be raised.

I didn't think at the time that `json.tools` would be used for large files hence the choice I made (note that this only work when reading a json lines formatted file, if the file has a large single JSON document, it will still be fully loaded in memory).

I you think this is an issue, I think another way around would be to use a temporary file to write to and to put it in its final destination once all writing is finished.

What do you think about this?